File: apps/site/snippets/perf/accessibility-label.kt
package snippets.perf

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.text.BasicText
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import dev.sinua.view.generated.SinuaOrb
import dev.sinua.view.generated.SinuaOrbPattern

@Composable
fun LabelledOrb() {
    Column {
        // An image with this name. Default: the spec's `name`, else the pattern.
        SinuaOrb(pattern = SinuaOrbPattern.LISTENING, contentDescription = "Assistant is listening", modifier = Modifier.size(160.dp))
        // Decorative (the text says it already): "" hides it from TalkBack.
        SinuaOrb(pattern = SinuaOrbPattern.LISTENING, contentDescription = "", modifier = Modifier.size(24.dp))
        BasicText("Listening…")
    }
}

File: apps/site/snippets/perf/accessibility-label.swift
import SwiftUI
import Sinua

struct LabelledOrb: View {
    var body: some View {
        VStack {
            // An image with this name. Default: the spec's `name`, else the pattern.
            SinuaOrb(pattern: .listening, accessibilityLabel: "Assistant is listening")
                .frame(width: 160, height: 160)
            // Decorative (the text says it already): "" hides it from VoiceOver.
            SinuaOrb(pattern: .listening, accessibilityLabel: "")
                .frame(width: 24, height: 24)
            Text("Listening…")
        }
    }
}
